pandas/core/categorical.py

```diff
@@ -118,6 +118,10 @@ def maybe_to_categorical(array):
 
 Assigning to `categories` is a inplace operation!
 
+Note
+----
+Categories getter returns all categories and not only those present in the data
```

Comment:
@Dorozhko-Anton current state looks good! Can you just squash the commits into one?